core/git.go

```diff
@@ -409,15 +809,25 @@ func (ref *LocalGitRef) PBDefinitions(ctx context.Context) ([]*pb.Definition, er
 	return ref.Repo.PBDefinitions(ctx)
 }
 
-func (ref *LocalGitRef) Tree(ctx context.Context, srv *dagql.Server, discardGitDir bool) (_ *Directory, rerr error) {
+func (ref *LocalGitRef) Tree(ctx context.Context, srv *dagql.Server, discardGitDir bool, depth int) (_ *Directory, rerr error) {
 	op, ok := DagOpFromContext[FSDagOp](ctx)
 	if !ok {
 		return nil, fmt.Errorf("no dagop")
 	}
-	bkref, err := op.CreateRef(ctx, nil,
+
+	commit, fullref, err := ref.Resolve(ctx)
+	if err != nil {
+		return nil, err
+	}
+	cacheKey := dagql.CurrentID(ctx).Digest().Encoded()
+
+	locker := ref.Query.Locker()
+	locker.Lock(gitSnapshotIndex + cacheKey)
+	defer locker.Unlock(gitSnapshotIndex + cacheKey)
+	bkref, err := op.Cache().New(ctx, nil, op.Group(),
```

Comment:
Why do we have this lock but then unconditionally always create+return a new ref? I would have expected the lock to be useful when it's "get-or-init"-type behavior

Reply:
Reasonable point. There's no need for a lock here :tada: